sky/data/storage.py

```diff
@@ -1871,6 +1904,542 @@ def _delete_gcs_bucket(self, bucket_name: str) -> bool:
+                                        }
+                                    },
+                                    'key_source': 'Microsoft.Storage'
+                                },
+                            }).result()
+                        # Assigning Storage Blob Data Owner role of the storage account
+                        # Reference: https://github.com/Azure/azure-sdk-for-python/issues/35573 # pylint: disable=line-too-long
+                        authorization_client = data_utils.create_az_client(
+                            'authorization')
+                        graph_client = data_utils.create_az_client('graph')
+                        async def get_object_id():
+                            import logging
+                            original_level = logging.getLogger("httpx").getEffectiveLevel()
+                            # silencing the INFO level response log from httpx request
+                            logging.getLogger("httpx").setLevel(logging.WARNING)
+                            user = await graph_client.users.with_url(
+                                'https://graph.microsoft.com/v1.0/me').get()
+                            logging.getLogger("httpx").setLevel(original_level)
+                            object_id = str(user.additional_data['id'])
+                            return object_id
+                        object_id = asyncio.run(get_object_id())
+                        # Defintion ID of Storage Blob Data Owner role.
+                        # Reference: https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles/storage#storage-blob-data-owner # pylint: disable=line-too-long
+                        storage_blob_data_owner_role_id = 'b7e6dc6d-f1e8-4753-8033-0f276bb0955b'
+                        role_definition_id = ('/subscriptions'
+                                            f'/{azure.get_subscription_id()}'
+                                            '/providers/Microsoft.Authorization'
+                                            '/roleDefinitions'
+                                            f'/{storage_blob_data_owner_role_id}')
+                        authorization_client.role_assignments.create(
+                                scope=creation_response.id,
+                                role_assignment_name=uuid.uuid4(),
+                                parameters={
+                                    "properties": {
+                                        "principalId": f'{object_id}',
+                                        "principalType": "User",
+                                        "roleDefinitionId": role_definition_id,
+                                    }
+                                },
+                            )
+                    except azure.exceptions().ResourceExistsError as e:
+                        error_message = e.message
+                        if 'StorageAccountAlreadyTaken' in error_message:
```

Comment:
Also, is it possible to infer the resources group for an existing storage account, if the resources group is not specified in the skypilot config? Currently it will use the default resources group name and raise an error saying the storage account not found in the default resources group name.

Reply:
1.  > which is due to no storage account found in the resources group (since it is in another resources group).

Added additional error handling you mentioned in the above comment: [144a60d](https://github.com/skypilot-org/skypilot/pull/3032/commits/144a60d2bb2824759a1ff43defa2fee3e751dffd)

2. > Also, is it possible to infer the resources group for an existing storage account

This issue should be addressed as we make the decision on https://github.com/skypilot-org/skypilot/pull/3032#discussion_r1628565409

Reply:
Humm, for 1, I think the problem is that we need to re-raise for **all possible exceptions**, but not only 2 (as implemented now) or 1 (as implemented previously) type of exceptions, since in either case it means a creation of the storage account failed. And in later code, we assume that the storage account always exists, which is bug-prone.

<img width="912" alt="image" src="https://github.com/skypilot-org/skypilot/assets/74357442/5acc8a2c-c87b-4293-ba1e-88a50a76bd21">

Reply:
Thanks for the elaboration! Added another general exception handling at [a1940a2](https://github.com/skypilot-org/skypilot/pull/3032/commits/a1940a257cd3a041fbbfbbb20fc279568d2427a5)

Reply:
Great! 🚀